File: app/src/main/java/bj/drunkmessengerblocker/BaseActivity.java
package bj.drunkmessengerblocker;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

/**
 * Created by j on 15/01/2017.
 */

public abstract class BaseActivity extends AppCompatActivity
{
    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setupComponent(App.component());
    }

    protected abstract void setupComponent(AppComponent appComponent);
}

File: app/src/main/java/bj/drunkmessengerblocker/common/BaseActivity.kt
package bj.drunkmessengerblocker.common

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import bj.drunkmessengerblocker.App
import bj.drunkmessengerblocker.AppComponent

/**
 * Created by Josh Laird on 23/05/2017.
 */
abstract class BaseActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setupComponent(App.component())
    }

    abstract fun setupComponent(appComponent: AppComponent)
}